app/infrastructure/queue_consumer.py

```
@@ -154,30 +244,118 @@ async def _process_job(self, worker_id: str, job_data: Dict[str, Any]):
                 await asyncio.sleep(30)  # Report every 30 seconds
             except Exception as e:
                 self.logger.error(f"Stats reporter error: {e}")
                 await asyncio.sleep(30)
+    
+    async def _notify_permanent_failure(self, job_tracer):
+        try:
+            
+            serialized_model = job_tracer.model_dump()
+            
+            if not settings.mail.MAIL_AUDIT_RECIPIENTS:
+                raise RuntimeError("MAIL_AUDIT_RECIPIENTS is not configured")
```

Ruff (0.13.3)

Abstract `raise` to an inner function

(TRY301)